straxen/scada.py

```diff
@@ -0,0 +1,275 @@
+import urllib
+import requests
+import pandas as pd
+import numba
+import numpy as np
+import warnings
+
+import strax
+import straxen
+
+export, __all__ = strax.exporter()
+
+
+@export
+class SCADAInterface:
+
+    def __init__(self, context=None):
+        """
+        Interface to excess the XENONnT slow control data via python.
+
+        :param context: Context you are using e.g. st. This is needed
+            if you would like to query data via run_ids.
+        """
+        try:
+            self.SCData_URL = straxen.get_secret('SCData_URL')
+            self.SCLastValue_URL = straxen.get_secret('SCLastValue_URL')
+            self.SCADA_SECRETS = straxen.get_secret('SCADA_SECRETS')
+        except ValueError:
+            raise ValueError('Cannot load SCADA information, from xenon'
+                             ' secrets. SCADAInterface cannot be used.')
+        self.st = context
+
+    def get_scada_values(self,
+                         parameters,
+                         start=None,
+                         end=None,
+                         run_id=None,
+                         time_selection_kwargs={'full_range': True},
+                         value_every_seconds=1):
+        """
+        Function which returns XENONnT slow control values for a given set
+        of parameters and time range.
+
+        The time range can be either defined by a start and end time or via
+        the run_id, target and context.
+
+        :param parameters: dictionary containing the names of the requested
+            scada-parameters. The keys are used as identifier of the
+            parameters in the returned pandas.DataFrame.
+        :param start: int representing the start time of the interval in ns
+            unix time.
+        :param end: same as start but as end.
+        :param run_id: Id of the run. Can also be specified as a list or
+            tuble of two run ids. In this case we will return the time
+            range lasting between the start of the first and endtime of the
+            second run.
+        :param time_selection_kwargs: Keyword arguments taken by
+            st.to_absolute_time_range(). Default: full_range=True.
+        :param value_every_seconds: Defines with which time difference
+            values should be returned. Must be an integer!
+            Default: one value per 1 seconds.
+        :return: pandas.DataFrame containing the data of the specified
+            parameters.
+        """
+        if not isinstance(parameters, dict):
+            mes = 'The argument "parameters" has to be specified as a dict.'
+            raise ValueError(mes)
+
+        if np.all((run_id, self.st)):
+            # User specified a valid context and run_id, so get the start
+            # and end time for our query:
+            if isinstance(run_id, (list, tuple)):
```

Comment:
Why not string?

Reply:
Because, this is only needed if the user specifies a list/tuple of run_ids.